Pull RSS feed from blog for events

endorama opened the pull request:
<!--
Thanks for opening a pull request!

Please remember to:
- mention any issue(s) that this PR closes using a closing keyword as well as the issue number, such as "Closes #XYZ" or "Resolves sigstore/repo-name#XYZ", cf.
  [documentation](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- ensure your commits are signed-off, as sigstore uses the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin) using `git commit -s`, or `git commit -s --amend` if you want to amend already existing commits
- lastly, ensure there are no merge commits!
Thank you :)
-->

#### Summary
<!--
A description of what this pull request does, as well as QA test steps (if applicable and if not already added to the Jira ticket).
-->

Add logic to events page that pulls RSS information from blog.sigstore.dev

There are some implementation choices open for discussion:
- using `axios`, for fetching content
- leveraging `api.rss2json.com`, to transform XML feed to JSON
- changing `date` to `pubDate` in events markdown frontmatters, to align with data from the RSS feed and allow sorting more easily

Not sure is this PR needs a release note (PR is editable by maintainers in case).

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/sigstore/YYYYYY/issues/XXXXX

-->
Closes https://github.com/sigstore/sigstore-website/issues/114

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates)
* API additions—new endpoint, new response fields, or newly accepted request parameters
* Database changes (any)
* Websocket additions or changes
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating)
* New features and improvements, including behavioural changes, UI changes and CLI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense.

-->
```release-note

```

bobcallaway commented:
Looks great, thanks @endorama !

Two minor questions: 
1. Should we limit the list to the last 5 posts?
2. I noticed in one of the titles we had `&amp;` instead of `&`... is there some escaping we need to do on the titles?

mdunbavan commented:
@bobcallaway @endorama I think the limit to 5 posts is a great idea. When we initially built this we had that setup as a possibility but there wasn't scope for lots of articles being written but now sigstore is gaining traction articles are coming in thick and fast :)

Great work btw @endorama

mdunbavan commented on components/newsEvents.vue:
```diff
@@ -70,10 +70,30 @@ export default {
+            } catch {
+                console.error("Cannot retrieve blog RSS feed");
+            }
+
```
@endorama I think pagination should reside in here somewhere

mdunbavan reviewed (changes requested):
If we add the pagination I think we are good to go

endorama commented:
Thanks all!

> Should we limit the list to the last 5 posts?

I think for sure in the home page, but what about in the community page? Should we add pagination to both or limit to 5 for home page and paginate (maybe at 10) in community page?

> I noticed in one of the titles we had &amp; instead of &... is there some escaping we need to do on the titles?

Probably the XML to JSON conversion doing weird things, I'll have a look.

endorama commented:
I added pagination, the page number style is very basic, comments welcome! I have to refine it a bit (as there is no logic limiting number of displayed page this will go wild when there will a lot of articles) but the general implementation and design is there.
I added it only to the community page. 

What do you think?

endorama commented:
I was also wondering if there was a way to fetch the feed list at build time instead that requesting them for each visitor (not our APIs but free version is limited + lot of useless traffic). I think so but my experience with NuxtJS was zero before this PR. so I'm not really a pro :)

bobcallaway commented:
LGTM - I love how it folds in the RSS feed with the ability to still specify manual news stories.

lukehinds commented:
@mdunbavan this still has you as 'requested changes', is this good now?
